workflows/model_retraining_workflow/train_model_workflow.py

```diff
@@ -0,0 +1,184 @@
+import datetime
+from functools import partial
+import os
+from typing import List, Tuple
+from truefoundry.workflow import (
+    task,
+    workflow,
+    map_task,
+    PythonTaskConfig,
+    TaskPythonBuild,
+    ExecutionConfig,
+)
+from truefoundry.deploy import Resources
+import numpy as np
+
+from train_models import train_respective_model
+
+task_config = PythonTaskConfig(
+    image=TaskPythonBuild(
+        python_version="3.10",
+        pip_packages=["truefoundry[workflow]>=0.5.1"],
+        requirements_path="requirements.txt",
+    ),
+    resources=Resources(
+        cpu_request=1,
+        cpu_limit=1,
+        memory_request=1000,
+        memory_limit=1500,
+        ephemeral_storage_request=1000,
+        ephemeral_storage_limit=1500,
+    ),
+    env={
+        "TFY_API_KEY": "<Your-pat-token>",
+        "TFY_HOST": "<host>",
+        "WHYLABS_DEFAULT_ORG_ID": "<org-id>",
+        "WHYLABS_DEFAULT_DATASET_ID": "<dataset-id>",
+        "WHYLABS_API_KEY": "<api-key>",
+    },
+)
+
+
+@task(task_config=task_config)
+def check_drift() -> bool:
+    import random
+
+    drift_value = random.randint(1, 10)
+    if drift_value > 5:
+        print("Drift is detected in the data")
+        return True
+    print("No drift detected in the data")
+    return False
+
+
+@task(task_config=task_config)
+def load_and_preprocess_data(
+    is_drift_present: bool,
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    if not is_drift_present:
+        print("No drift detected in the data, skipping data preprocessing")
+        return (np.array([]), np.array([]), np.array([]), np.array([]))
+    import pandas as pd
+    from sklearn.model_selection import train_test_split
+
+    df = pd.read_csv("data.csv")
```

Comment:
Add a note here that in case of drift the latest data will need to be fetched from external systems. We are just using the data.csv for demo purpose